spec.html

```diff
@@ -787,68 +787,97 @@ <h1>Throw an Exception</h1>
 
       <emu-clause id="sec-returnifabrupt" aoid="ReturnIfAbrupt">
         <h1>ReturnIfAbrupt</h1>
-        <p>Algorithms steps that say or are otherwise equivalent to:</p>
+        <p>An algorithm step that says:</p>
         <emu-alg>
-          1. ReturnIfAbrupt(_argument_).
+          1. *[before]* ReturnIfAbrupt(_argument_) *[after]*
         </emu-alg>
-        <p>mean the same thing as:</p>
+        <p>is equivalent to the sequence of steps:</p>
         <emu-alg>
           1. If _argument_ is an abrupt completion, return _argument_.
-          1. Else if _argument_ is a Completion Record, set _argument_ to _argument_.[[Value]].
+          1. Else, if _argument_ is a Completion Record, set _argument_ to _argument_.[[Value]].
+          1. *[before]* _argument_ *[after]*
         </emu-alg>
-        <p>Algorithms steps that say or are otherwise equivalent to:</p>
+        <p>where</p>
+        <ul>
+          <li>*[before]*, if present, is any algorithm text that does not contain ReturnIfAbrupt.
+          <li>*[after]*, if present, is any algorithm text.
+          <li>_argument_ is any variable.
+        </ul>
+
+        <p>An algorithm step that says:</p>
         <emu-alg>
-          1. ReturnIfAbrupt(AbstractOperation()).
+          1. *[before]* ReturnIfAbrupt(AbstractOperation(*[arguments]*)) *[after]*
```

Comment:
Rather than having 2 definitions, discriminated by the form of the argument, you could have one definition with a generalized argument.

Reply:
Since it's a structural rewrite, I'd like to keep it as simple as I can. Though one could argue that 1 rewrite rule is simpler overall than 2 individually simpler rewrite rules. I'll leave that one up to @tc39/ecma262-editors.

Reply:
1 seems probably simpler, but totes fine to do that in a followup.

Reply:
I've reduced the ReturnIfAbrupt definition to a single macro expansion, though it now has the peculiarity that, when used on an alias, as in `ReturnIfAbrupt(_someAlias_)`, we are left with a step at the end that just references the alias. I think that's probably fine, but it may not be clear to everyone that that is meant to be a no-op. Let me know if you want me to switch it back to 2 expansions, I'm not tied to a particular choice.

Reply:
I like this as a single expansion. I think the "step at the end that just references the alias" thing is unlikely to trip up readers, but if we're worried, we could say "the last line is omitted if `*[before]*` is not present" (or "is empty" or whatever).